autoload/neomake/makers/ft/typescript.vim

```diff
@@ -7,8 +7,9 @@ endfunction
 function! neomake#makers#ft#typescript#tsc()
     return {
         \ 'args': [
-            \ '-m', 'commonjs', '--noEmit'
+            \ '--project', getcwd(), '--noEmit'
```

Comment:
No TSC will traverse up the directory tree until it finds the first tsconfig.json. I'll try what you suggested